.github/workflows/UnitTests.yml

```diff
@@ -49,6 +49,20 @@ jobs:
     - name: Analysing the code with pylint
       run: |
         pylint MaxText/
+    - name: Cleanup old docker images
+      run: |
+        docker system prune --all --force
+    - name: Install dependencies
+      run: |
+        bash docker_build_dependency_image.sh
+    - name: Test standalone_dataloader.py
+      run: |
+        docker run -v /home/runner/actions-runner/_work/maxtext/maxtext:/app --rm --privileged maxtext_base_image bash -c \
+        'python3 MaxText/standalone_dataloader.py MaxText/configs/base.yml run_name=standalone_dataloader_$(date +%Y-%m-%d-%H-%M)-${RANDOM} base_output_directory=gs://runner-maxtext-logs dataset_path=gs://maxtext-dataset steps=100 enable_checkpointing=false hardware=cpu'
+    - name: Test standalone_checkpointer.py checkpoint at 50 and restore at 50
+      run: |
+        docker run -v /home/runner/actions-runner/_work/maxtext/maxtext:/app --rm --privileged maxtext_base_image bash -c \
```

Comment:
Fair thought, but I want them to run before a PR is submitted. This will protect the checkpointer from incoming changes such as altering the PyTree format or altering the restore mechanism, etc. WDYT?

Reply:
The tests under [tests](https://github.com/google/maxtext/tree/main/MaxText/tests) are also run every pull request via [pytest](https://github.com/google/maxtext/blob/3ab19a0baef5b786a333f5ddf199560a841d7c45/.github/workflows/UnitTests.yml#L74-L76)

This was more of a nit, its okay to leave here.  I wanted to move it so its more readable and easier to maintain.

Reply:
Thanks Matt, I handle this in an upcoming PR. I am also considering moving some tests to Airflow-XL-ML .

Reply:
Moved the standalone dataloader and checkpointer tests for TPUs to pytest, as recommended. PTAL!